### PR TITLE
ISSUE-679 [Deleted message vault] JMAP - Guice bindings + Integration tests

### PR DESCRIPTION
**Pull request by vttranlina**
resolve https://github.com/linagora/tmail-backend/issues/679

### CODE REVIEW

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/EmailRecoveryActionIntegrationTest.scala [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+             |      }, "c1"]] }""".stripMargin)
+      .when()
+        .post()
+      .`then`
+        .body("methodResponses[0][1].ids", Matchers.empty())
+    })
+  }
+
+  @Test
+  def recoveryShouldNotBeRemovedFromTheVault(): Unit = {
```
Comment (Arsnael) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/EmailRecoveryActionIntegrationTest.scala:133]: Open question: should an email really not be removed after we recovered it?

Comment (chibenwa) [reply tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/EmailRecoveryActionIntegrationTest.scala:133]: I think we had the debate once in a loong loong time, can't recall the answer.

Comment (vttranlina) [reply tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/EmailRecoveryActionIntegrationTest.scala:133]: Your mean is we should remove the vault after recovery? or this test is not ensured?
It is based on the `org.apache.james.webadmin.integration.vault.DeletedMessageVaultIntegrationTest#restoredMessagesShouldNotBeRemovedFromTheVault`

Comment (Arsnael) [reply tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/EmailRecoveryActionIntegrationTest.scala:133]: If it was the original design then (thus the loooong looooong time ago^^) I'm fine with this then. Thanks for the precision :)

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Pull request merged**